lib/timescale/query_utils.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Timescale.QueryUtils do
+  @moduledoc false
+
+  @doc """
+  This function generates a dynamic fragment based upon the optional
+  arguments that are provided. If no optional arguments are provided,
+  only the required arguments are passed to the TimescaleDB function
+  fragment.
+
+  For example, calling `Timescale.Hyperfunctions.time_bucket/3` with optional
+  arguments would yield something like so:
+
+  ```elixir
+  from(
+    t in Table,
+    select: time_bucket(t.timestamp, "5 minutes", origin: "1900-01-01", timezone: "Europe/Athens")
+  )
+
+  Would yield the following SQL query:
+
+  ```
+  SELECT time_bucket('5 minutes', m0."timestamp", origin => '1900-01-01', timezone => 'Europe/Athens') FROM "my_table" AS m0
+  ```
+  """
+  @spec dynamic_function_fragment(
+          function_name :: atom(),
+          required_args :: list(),
+          optional_arg :: keyword()
+        ) :: Macro.t()
+  def dynamic_function_fragment(function_name, required_args, optional_args) do
+    # TODO: Perhaps we also match on: `{named_argument, {_value, type}}`
```

Comment:
Thoughts on this one @davydog187?

Reply:
I love that idea, but maybe we leave it for another PR since there might be some complexity to it that i'm not thinking through. WDYT?

Reply:
That sounds good. Opened up this issue: https://github.com/bitfo/timescale/issues/14